File: Shapes/app/src/main/java/com/example/shapes/MainActivity.java
package com.example.shapes;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.appcompat.widget.AppCompatButton;

import android.os.Bundle;
import android.util.Log;
import android.view.View;

public class MainActivity extends AppCompatActivity implements DialogFragment.OnNegativeButtonClick, DialogFragment.OnPositiveButtonClick {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);

        AppCompatButton exist_button = findViewById(R.id.exist_button);
        exist_button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                DialogFragment dialogFragment = DialogFragment.newInstance("Confirmation",
                        "are you sure you wants to leave the app?",R.drawable.warning_img);
                dialogFragment.show(getSupportFragmentManager(),"dialogFragment");
            }
        });

        AppCompatButton change_button = findViewById(R.id.change_button);
        change_button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                try {
                    change_button.setText(null);
                    throw new Exception("show this log message when app crashes");
                }
                catch (Exception e){
                    Log.e("buttonErr",e.getMessage());
                }
            }
        });
    }

    @Override
    public void onClickPositiveButton() {
        this.finish();
    }

    @Override
    public void onClickNegativeButton() {

    }

}

File: PrintNum/PrintNum7/app/src/main/java/com/example/printnum7/MainActivity.kt
package com.example.printnum7

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        resultButton.setOnClickListener{
            val start7 = startEditText7.text.toString().toInt()
            val end7 = endEditText7.text.toString().toInt()
            var count7=0
            for (num in start7..end7){
                if(num%7 == 0){
                    count7++
                }
            }
            resultTextView7.text="Output: $count7"
            Log.d("result1",resultTextView7.text.toString())
            val start3 = startEditText3.text.toString().toInt()
            val end3 = endEditText3.text.toString().toInt()
            var count3=""
            for (num in start3..end3){
                if(num%3 == 0 && num%5 == 0){
                    break
                }
                if(num%3 != 0){
                    count3+= "$num "
                }
            }
            resultTextView3.text="Output: $count3"
            Log.d("result2",resultTextView3.text.toString())
        }
    }
}
